src/views/ChooseAddress.vue

```diff
@@ -62,9 +83,39 @@ export default class ChooseAddress extends Vue {
             userFriendlyAddress: accountOrContractInfo.userFriendlyAddress,
         });
 
-        const result: Address = {
+        let btcAddress: string | undefined;
+
+        if (this.request.returnBtcAddress && walletInfo.btcXPub) {
+            this.state = this.State.SYNCING;
+            // const startIndex = Math.max(Math.min(
+            //     walletInfo.btcAddresses.external.findIndex((addressInfo) => !addressInfo.used),
+            //     walletInfo.btcAddresses.internal.findIndex((addressInfo) => !addressInfo.used),
+            // ), 0);
```

Comment:
Not scanning all addresses again would indeed be good.
The current implementation tries to fill gaps between used addresses, which is good and wouldn't be accomplished by just using the start index of the last used address.
However, scanning could start at the last known, gap-less used address for internal and external addresses or even better just re-scan only the known, unused addresses for whether they have been used in the meantime and start scanning at the last known address only if then still required.